README.md

```diff
@@ -1,2 +1,555 @@
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+@cli
+def entry(x: Arg = 1, y: Arg):
+    print(x, y)
+
+if __name__ == '__main__':
+    cli()
+```
+
+The function signature is not supported even in Python.
+
+#### `Flag`
+
+`Flag` is an annotation for flags.
+In CLI, flags are boolean arguments that represent a turned off or turned on behavior.
+Unlike positional ones, they could be specified in a command line only with a special syntax.
+
+For example, the following function signature:
+```py
+# test.py
+
+from {pyname} import cli, Flag
+
+@cli
+def entry(some: Flag):
+    print(some)
+
+if __name__ == '__main__':
+    cli()
+```
+
+Could be called from the command line as follows:
+```
+$ python test.py --some
+True
+```
+
+But it's also possible to invoke the CLI without the flag:
+```
+$ python test.py
+False
+```
+
+Flags also support default values, but that doesn't make much sense.
+The default value of `False` is already set implicitly for each flag.
+The default value of `True` makes a flag to be always true.
+
+For example, one could define:
+```py
+# test.py
+
+from {pyname} import cli, Flag
+
+@cli
+def entry(some: Flag = True):
+    print(some)
+
+if __name__ == '__main__':
+    cli()
+```
+
+But this makes the flag `some` useless:
+```
+$ python test.py --some
+True
+$ python test.py
+True
+```
+
+#### `Opt`
+
+`Flag` is an annotation for options.
+In CLI, options are simply named arguments or flags with values.
+
+For example, the following function signature:
+```py
+# test.py
+
+from {pyname} import cli, Opt
+
+@cli
+def entry(some: Opt):
+    print(some)
+
+if __name__ == '__main__':
+    cli()
+```
+
+Could be called from the command line as follows:
+```
+$ python test.py --some 1
+1
+```
+
+Note: unlike with flags, it's not possible to _not_ specify the option by default.
+```
+$ python test.py --some
+usage: entry [-h] --some SOME
+entry: error: the following arguments are required: --some
+```
+
+But it's possible to specify the default value for `Opt`:
+```py
+# test.py
+
+from {pyname} import cli, Opt
+
+@cli
+def entry(some: Opt = 1):
+    print(some)
+
+if __name__ == '__main__':
+    cli()
+```
+
+So it's now possible to call the script as follows:
+```
+$ python test.py
+1
+```
+
+### Help page
+
+Each CLI in {Name} has a built-in help page, which is automatically generated.
+
+For example, for the following function:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+@cli
+def entry(some: Arg):
+    print(some)
+
+if __name__ == '__main__':
+    cli()
+```
+
+It's possible to invoke the help page using the flag `--help`:
+```
+$ python test.py --help
+usage: entry [-h] some 
+
+positional arguments:
+  some
+
+optional arguments:
+  -h, --help  show this help message and exit
+```
+
+Note the lack of the overall program description as well as the description of the `some` argument.
+
+To override the program description, one could write a simple doc-comment for the CLI function:
```

Comment:
I believe that there should be `.` in the end.

Reply:
Fixed!